src/main/java/org/apache/commons/configuration2/AbstractConfiguration.java

```diff
@@ -419,6 +419,20 @@ public final boolean containsKey(final String key) {
         }
     }
 
+    /**
+     * {@inheritDoc} This implementation handles synchronization and delegates to {@code containsKeyInternal()}.
+     */
+    @Override
+    public final boolean containsValue(final String value) {
+        Objects.requireNonNull(value, "Parameter \"value\" cannot be null");
```

Comment:
- Replace API calls of this pattern with simply:
`Objects.requireNonNull(value, "value");`
See the rest of the code base.
- Why does this check need to happen here at all? After all I don't get an NPE when I ask for a null value from a `HashMap`, for example ` new HashMap().containsValue(null)` returns false.

Reply:
containsValue delegates to `containsKeyInternal()`.

`containsKeyInternal()` implementation can be different depending on the implementing class.

If null is unavoidable we should always encourage safe null handling practices, such as using Objects.requireNonNull or conditional checks, to minimize the risk of (unexpected, unmanaged or uncontrolled) null pointer exceptions.

While `new HashMap().containsValue(null)` returns false, this implementation is inspired on `java.util.HashTable.contains()` and not HashMap which is also `java.util.Properties` implementation (extends HashTable).

```java
    // HashTable contains Implementation checks null at entry point
    public synchronized boolean contains(Object value) {
        if (value == null) {
            throw new NullPointerException();
        }

        Entry<?,?> tab[] = table;
        for (int i = tab.length ; i-- > 0 ;) {
            for (Entry<?,?> e = tab[i] ; e != null ; e = e.next) {
                if (e.value.equals(value)) {
                    return true;
                }
            }
        }
        return false;
    }
```

I find this to be a much more coherent implementation since it doesn't fail silently and explicitely warns the user that the input value is null.

The reason for the "custom" message is so we can control exactly how containsValue fails.

What do you think?

Reply:
After discussion with @Claudenw I have removed the null validation. Please see comments.

Thank you :)